WoW-Realm-Tracker/Models/Realm.swift

```diff
@@ -1,27 +1,36 @@
 import Runes
 import Argo
+import Curry
 
 struct Realm {
-    let name: String
-    let type: String
-    let status: Bool
+    enum RealmType: String, CustomStringConvertible {
+        case PvP = "pvp"
```

Comment:
Note: This may be a terrible idea :)

Hmm, what about having a method to convert the value when you parse it so you only have to worry about it there?

```swift
func apiTypeToOurs(type: String) -> Decoded<String> {
    return .fromOptional(string.toDate(format: .ISO8601))
    var capitalized: String?

    switch self {
      case "pvp": capitalized = "PvP"
      case "pve": capitalized = PvE"
      case "rp": capitalized = "RP"
      case "rppvp": capitalized = "RP PvP"
      default: break
    }

    return .fromOptional(capitalized)
}

// and

<*> (j <| "type" >>- apiTypeToOurs)
```